example.py

```diff
@@ -16,6 +16,8 @@ async def main() -> None:
         batch_count += 1
         row_count += len(batch)
         print(f"Processed batch {batch_count} with {len(batch)} rows")
+        await asyncio.sleep(0.01)
```

Comment:
Is this needed? If so, maybe a comment would be good.

Reply:
Sure, I'll add a comment! It's to artificially slow down processing so you can better see the async nature of read and process.